// winbox/src/js/plugin/storage.js
var storage = localStorage;
var index = {};
var records = [];

export default function(self){

    const old_fn = self.onclose;

    self.onclose = function(force){

        if(old_fn && old_fn(force)){

            return true;
        }


    }
}

function openWindow(title, options){

    if(typeof title === "object"){

        options = title;
        title = null;
    }

    options || (options = {});

    var onclose = options.onclose;
    options.onclose = function(){

        records.splice(index[this.id], 1, 0);
        delete index[this.id];
        onclose && onclose();
        save();
    };

    var winbox = new WinBox(title, options);

    index[winbox.id] = records.length;

    records.push({

        title: title,
        options: options
    });

    save();
}

function restore(){

    var data = storage.getItem("winbox");

    if(data){

        data = JSON.parse(data);

        for(let i = 0; i < data.length; i++){

            openWindow(data[i].title, data[i].options);
        }
    }
}

function save(){

    storage.setItem("winbox", JSON.stringify(records));
}

function clear(){

    storage.removeItem("winbox");
    records = [];
}
